### PR TITLE
Add DelegatorActions trait

### PR DESCRIPTION
**Pull request by nvengal**
Exposes `delegator_bond_more` to be called by external pallets (automation-time).

### CODE REVIEW

File: pallets/parachain-staking/src/lib.rs [hunk 1]
```diff
@@ -1682,6 +1679,40 @@ pub mod pallet {
...
+			ensure!(
+				!Self::delegation_request_revoke_exists(candidate, delegator),
+				Error::<T>::PendingDelegationRevoke
+			);
+			let mut state = <DelegatorState<T>>::get(delegator).ok_or(Error::<T>::DelegatorDNE)?;
+			state.increase_delegation::<T>(candidate.clone(), more)?;
+			Ok(())
+		}
+
+		fn testing_setup_delegator(
```
Comment (nvengal) [inline pallets/parachain-staking/src/lib.rs]: I added this solely to be able to write benchmarks from automation-time that include staking functionality. I originally tried to feature flag this such that this method only compile while running tests but unfortunately that's not yet a supported feature- https://github.com/rust-lang/cargo/issues/2911

Open to any other patterns or ideas for achieving this.

File: pallets/parachain-staking/src/lib.rs [hunk 2]
```diff
@@ -1682,6 +1679,40 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> DelegatorActions<T::AccountId, BalanceOf<T>> for Pallet<T> {
+		fn delegator_bond_more(
+			delegator: &T::AccountId,
+			candidate: &T::AccountId,
+			more: BalanceOf<T>,
+		) -> DispatchResult {
```
Comment (justinzhou93) [inline pallets/parachain-staking/src/lib.rs]: how come DispatchResult even though the test setup delegator returns DispatchResultWithPostInfo? DispatchResultWithPostInfo seems to include the weight of the action if i read the docs correctly? do we need that weight eventually?

**Review by justinzhou93 — APPROVED** [review]
2 questions otherwise lgtm!
1. in regards to the return type, see line comment for more details
2. to make sure i'm understanding the reason properly you're just refactoring into a function to create a trait so that you can stub interaction in the OAK blockchain repo, right?

File: pallets/parachain-staking/src/lib.rs (hunk 1 above)
Comment (rhuttman) [reply pallets/parachain-staking/src/lib.rs]: If its just for benchmarks you could add the benchmark compile flag. The difference being you already need to compile your code with that flag for benchmarks

**Review by rhuttman — APPROVED** [review]

File: pallets/parachain-staking/src/lib.rs (hunk 2 above)
Comment (nvengal) [reply pallets/parachain-staking/src/lib.rs]: Not sure if we'll need the weight or not but I went ahead and changed the return type of this function.

File: pallets/parachain-staking/src/lib.rs (hunk 1 above)
Comment (nvengal) [reply pallets/parachain-staking/src/lib.rs]: Great idea. That's exactly what I was looking for.

**Review by justinzhou93 — APPROVED** [review]

**Review by rhuttman — APPROVED** [review]

**Pull request merged**